boards/xiao_m0/Cargo.toml

```diff
@@ -16,13 +16,15 @@ optional = true
 
 [dependencies.atsamd-hal]
 version = "0.14"
+path = "../../hal"
```

Comment:
I had to update the Hal to reference the repo version because a Monotonic related trait was not being satisfied. Feather M0 does the same in it's repo, I would assume for this same reason